### PR TITLE
[Fixes #7881] Add filtering for start_date, end_date, categories and keywords for geonode harvester

### PR DESCRIPTION
**Pull request by meomancer**
…eonode harvester

<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [ ] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [ ] The issue connected to the PR must have Labels and Milestone assigned
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

**Comment by meomancer** [comment]
This fix https://github.com/GeoNode/geonode/issues/7880

CC @ricardogsilva

File: geonode/harvesting/harvesters/geonode.py [hunk 1]
```diff
@@ -144,6 +161,20 @@ def get_extra_config_schema(cls) -> typing.Dict:
...
+                },
+                "end_date_filter": {
+                    "type": "string",
+                    "format": "date-time"
+                },
+                "keywords_filter": {
+                    "type": "array",
+                },
+                "categories_filter": {
+                    "type": "array",
```
Comment (ricardogsilva) [inline geonode/harvesting/harvesters/geonode.py:176]: Please add a section describing the items of the array

```suggestion
                "keywords_filter": {
                    "type": "array",
                    "items": {
                        "type": "string",
                    }
                },
                "categories_filter": {
                    "type": "array",
                    "items": {
                        "type": "string",
                    }
```

File: geonode/harvesting/harvesters/geonode.py [hunk 2]
```diff
@@ -393,6 +424,20 @@ def _get_resource_list_params(
...
         if self.resource_title_filter is not None:
             result["title__icontains"] = self.resource_title_filter
+        if self.start_date_filter is not None:
+            start_date = dateutil.parser.parse(self.start_date_filter)
+            result["date__gte"] = start_date.astimezone(
+                pytz.utc).replace(microsecond=0).isoformat().split('+')[0] + 'Z'
+        if self.end_date_filter is not None:
+            end_date = dateutil.parser.parse(self.end_date_filter)
+            result["date__lte"] = end_date.astimezone(
+                pytz.utc).replace(microsecond=0).isoformat().split('+')[0] + 'Z'
```
Comment (ricardogsilva) [inline geonode/harvesting/harvesters/geonode.py:434]: Please use Python's built in UTC support instead of pytz here

`datetime.timezone.utc`

**Review by ricardogsilva — CHANGES_REQUESTED** [review]
Looks good. Please make the requested changes, which are very small and then I'll be +1 for merging

File: geonode/harvesting/harvesters/geonode.py (hunk 1 above)
Comment (meomancer) [reply geonode/harvesting/harvesters/geonode.py]: updated @ricardogsilva

File: geonode/harvesting/harvesters/geonode.py (hunk 2 above)
Comment (meomancer) [reply geonode/harvesting/harvesters/geonode.py]: updated @ricardogsilva

**Comment by meomancer** [comment]
Could you check this @ricardogsilva ?

**Comment by meomancer** [comment]
hi @giohappy 
could we merge this?

**Comment by giohappy** [comment]
@meomancer I've updated the branch. I will merge it as soon as tests have passed again.

**Pull request merged**